tests/Feature/Booking/BookingTest.php

```diff
@@ -275,85 +280,142 @@ public function testInvalidBookingCantBeCreated()
     }
 
     /** @test */
-    public function testOverlappingBookings()
+    public function testOverlappingBookingsCannotBeCreatedOverlappingOnStart()
     {
-        $position = factory(Position::class)->create()->id;
-        $position2 = factory(Position::class)->create()->id;
+        $this->bypassRatingChecks();
 
         // Create two bookings for position 1
         factory(Booking::class)->create([
-            'position_id' => $position,
+            'position_id' => $this->position->id,
             'from' => '2019-08-10 14:00:00',
             'to' => '2019-08-10 15:00:00',
         ]);
 
-        $booking2 = factory(Booking::class)->create([
-            'position_id' => $position,
-            'from' => '2019-08-10 19:00:00',
-            'to' => '2019-08-10 20:00:00',
-        ]);
-
         // Test 1: Doesn't allow to book inside of booking
         $this->graphQL("
           mutation {
-            createBooking(user_id: 1300001, position: {
-                connect: {$position}
-            }, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
+            createBooking(user_id: {$this->mockUserId}, position_id: {$this->position->id}, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
                 id
             }
           }")->assertJsonPath('errors.0.message', "Can't have overlapping bookings for the same position!");
+    }
+
+    /** @test */
+    public function testOverlappingBookingsCannotBeCreatedOverlappingOnEnd()
```

Comment:
And this one overlaps on the start of the booking

Reply:
The one in the request starts before the created one.

Reply:
exactly, so it is overlapping the start, not the end, surely?! 😂

Reply:
(I think you might be looking at this from the bookings, i.e. the exiting one is overlapping on the end of the requested booking, perspective, were as I am looking at it from the perspective of the requesting booking is overlapping the existing one)